nixos/modules/security/krb5.nix

```
@@ -0,0 +1,153 @@
+  in "${name} = ${v}";
+
+  mkRemovedOptionModule' = name: reason: mkRemovedOptionModule ["krb5" name] reason;
+  mkRemovedOptionModuleCfg = name: mkRemovedOptionModule' name ''
+    The option `krb5.${name}' has been removed. Use
+    `security.krb5.settings.${name}' for structured configuration or
+    `security.krb5.extraConfig' for plain-text configuration.
+  '';
+  mkMovedOptionModule = name: mkRemovedOptionModule' name ''
+    The option `krb5.${name}' has been moved to `security.krb5.${name}'.
+  '';
+
+  cfg = config.security.krb5;
+in {
+  imports = [
+    (mkRemovedOptionModuleCfg "libdefaults")
+    (mkRemovedOptionModuleCfg "realms")
+    (mkRemovedOptionModuleCfg "domain_realm")
+    (mkRemovedOptionModuleCfg "capaths")
+    (mkRemovedOptionModuleCfg "appdefaults")
+    (mkRemovedOptionModuleCfg "plugins")
+    (mkRemovedOptionModule' "config" ''
+      The option `krb5.config' has been removed. Use `security.krb5.settings'
+      for structured configuration or `security.krb5.extraConfig' for plain-text
+      configuration.
+    '')
+    (mkMovedOptionModule "extraConfig")
+    (mkMovedOptionModule "kerberos")
+  ];
+
+  options = {
+    security.krb5 = {
+      enable = mkEnableOption (mdDoc "building krb5.conf, configuration file for Kerberos V");
+
+      kerberos = mkOption {
+        type = types.package;
+        default = pkgs.krb5;
+        defaultText = literalExpression "pkgs.krb5";
+        example = literalExpression "pkgs.heimdal";
+        description = lib.mdDoc ''
+          The Kerberos implementation that will be present in
+          `environment.systemPackages` after enabling this
+          service.
+        '';
+      };
+
+      settings = mkOption {
+        default = {};
+        description = mdDoc ''
+          Structured contents of the `krb5.conf` file. See `krb5.conf(5)`
+          for details about configuration.
+        '';
+        example = literalExpression ''
+          {
+            libdefaults = {
+              default_realm = "ATHENA.MIT.EDU";
+            };
+
+            realms = {
+              "ATHENA.MIT.EDU" = {
+                admin_server = "athena.mit.edu";
+                kdc = [
+                  "athena01.mit.edu"
+                  "athena02.mit.edu"
+                ];
+              };
+            };
+
+            domain_realm = {
+              "mit.edu" = "ATHENA.MIT.EDU";
+            };
+
+            logging = {
+              kdc = "SYSLOG:NOTICE";
+              admin_server = "SYSLOG:NOTICE";
+              default = "SYSLOG:NOTICE";
+            };
+          }
+        '';
+        type = toplevel;
+      };
+
+      extraConfig = mkOption {
+        type = lines;
+        default = "";
+        example = ''
+          includedir /etc/krb5.conf.d
+        '';
+        description = mdDoc ''
+          The verbatim contents of /etc/krb5.conf. `krb5.conf` may include any
+          of the relations that are valid for `kdc.conf` (see `kdc.conf(5)`),
```

Since it's a cleanup anyway

```suggestion
          The verbatim contents of {file}`/etc/krb5.conf`. {file}`krb5.conf` may include any
          of the relations that are valid for {file}`kdc.conf` (see {manpage}`kdc.conf(5)`),
```